### DISCUSSION TITLE
[QUESTION] how to expose the version of pybind11 that a package is built with (to be readable from Python)

### DISCUSSION THREAD

**Discussion by slayoo** [pybind/pybind11#4798] [Q&A] [answered]
In a pybind11-based project ([PyPartMC](https://pypi.org/p/PyPartMC/)), we have a couple of build-time dependencies that are either header-only C++ libs (incl. pybind11 itself) as well as some statically linked codes (C, Fortran, C++). All of the dependencies are linked as git submodules in the project repo. We are trying to figure out a best practice to make the versions (i.e., git hashes) of all these dependencies accessible from Python by the package users.

Something like:
```
import XXX as pkg
print(pkg.__version__)
print(pkg.__versions_of_build_time_dependencies__['pybind11'])
```

Any suggestions how to best construct it from the Python front? (i.e., how to name such version attributes, and would a dict suggested above be a good choice?)

Thanks,
Sylwester

**Comment by evguran** [comment, marked as answer]
I believe it is up to you to choice in which container to store the data, and what you suggested is good for a "versioning" of a big number of C++ libs. You named only few though, so in this case I would prefer string attributes `__version_of_pybind11__`  etc.

**Reply by slayoo** [reply]
Thanks! The key rationale for an iterable container would be to enable user code to iterate over it, e.g., to store the version information along the simulation output.

**Reply by slayoo** [reply]
for the record, here's how we applied it: https://github.com/open-atmos/PyPartMC/pull/310
